gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +89,101 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
     return ht
 
 
+def create_pangolin_grch38_ht(vcf_path: str) -> hl.Table:
+    """
+    Create a Hail Table with Pangolin score for splicing for GRCh38.
+
+    .. Note::
+    The Pangolin scores were generated for variants in gene body only. The first run contains variants from gnomAD v3 genomes.
+
+    :param vcf_path: path to the VCF files with Pangolin scores for splicing.
+    :return: Hail Table with Pangolin score for splicing for GRCh38.
+    """
+    mt = hl.import_vcf(vcf_path, min_partitions=1000, reference_genome="GRCh38")
+
+    ht = mt.rows()
+    logger.info("Number of rows in original Pangolin Hail Table: %s", ht.count())
+
+    logger.info("Exploding Pangolin scores...")
+    # `explode` will eliminate rows with empty array
+    ht = ht.annotate(pango=ht.info.Pangolin[0].split(delim="\|\|"))
+    ht = ht.explode(ht.pango)
+    logger.info("Number of rows in exploded Pangolin Hail Table: %s", ht.count())
+
+    logger.info("Annotating Pangolin scores...")
+    ht = ht.annotate(
+        pangolin=hl.struct(
+            gene=ht.pango.split(delim=":|\\|")[0],
+            positions=(
+                hl.empty_array(hl.tint32).append(
+                    hl.int(ht.pango.split(delim=":|\\|")[1])
+                )
+            ).append(hl.int(ht.pango.split(delim=":|\\|")[3])),
+            delta_scores=(
+                hl.empty_array(hl.tfloat64).append(
+                    hl.float(ht.pango.split(delim=":|\\|")[2])
+                )
+            ).append(hl.abs(hl.float(ht.pango.split(delim=":|\\|")[4]))),
+        )
+    )
+
+    logger.info(
+        "Getting the max Pangolin score for each variant across consequences per"
+        " gene..."
+    )
+    consequences = hl.literal(["splice_gain", "splice_loss"])
+    ht = ht.annotate(
+        pangolin=ht.pangolin.annotate(ds_max=hl.max(ht.pangolin.delta_scores))
+    )
+    ht = ht.annotate(
+        pangolin=ht.pangolin.annotate(
+            consequence_max=hl.if_else(
+                ht.pangolin.ds_max > 0,
+                consequences[ht.pangolin.delta_scores.index(ht.pangolin.ds_max)],
+                "no_consequence",
+            ),
+            position_max=hl.if_else(
+                ht.pangolin.ds_max > 0,
+                ht.pangolin.positions[
+                    ht.pangolin.delta_scores.index(ht.pangolin.ds_max)
+                ],
+                hl.missing(hl.tint32),
+            ),
+        )
+    )
```

Comment:
This is extremely similar (the same if I am not mistaken) to the splice_ai approach. I think it would make sense to create a function that is something like `process_splice_scores`.